nest-city-account/.env.example

```
@@ -1,44 +1,58 @@
-# Environment variables declared in this file are automatically made available to Prisma.
-# See the documentation for more detail: https://pris.ly/d/prisma-schema#accessing-environment-variables-from-the-schema
+# Example setup for local development
+# Key takeaways:
+# - uses staging cognito user pool - recommended to create separate user for your local database and for staging deployment
+# - the postgres, rabbitmq credentials are set to match docker-compose
+# - validation is written in a way where it does not pass with empty string (TODO this can be improved, along with the rest of the config setup)
 
-# Prisma supports the native connection string format for PostgreSQL, MySQL, SQLite, SQL Server, MongoDB and CockroachDB (Preview).
-# See the documentation for all the connection string options: https://pris.ly/d/connection-strings
+# Get the following secrets from team before starting development:
+# - AWS_COGNITO_SECRET
 
-# TURNSTILE_SECRET not set in development, dummy token is used
-# if you want to test captcha, add TURNSTILE_SECRET into your local .env
+# Get the other omitted secrets as needed when developing related sections of the service
 
-DATABASE_URL="postgresql://postgres:postgres@localhost:5432/city-account?schema=public"
+DATABASE_URL="postgresql://postgres:postgres@localhost:5422/city-account?schema=public"
 
+# aws
 AWS_COGNITO_CLIENT_ID="1pdeai19927kshpgikd6l1ptc6"
 AWS_COGNITO_USERPOOL_ID="eu-central-1_FZDV0j2ZK"
 AWS_COGNITO_REGION="eu-central-1"
 AWS_COGNITO_ACCESS="AKIA55KLWRXCW5KRHIYH"
-AWS_COGNITO_SECRET=""
+AWS_COGNITO_SECRET="!!!change-this!!!"
 
+# magproxy - get MAGPROXY_AZURE_CLIENT_SECRET if you need user verification
 MAGPROXY_URL="https://new-magproxy.bratislava.sk"
 
-
 MAGPROXY_AZURE_AD_URL="https://login.microsoftonline.com/fe69e74e-1e66-4fcb-99c5-58e4a2d2a063/oauth2/v2.0/token"
 MAGPROXY_AZURE_CLIENT_ID="ec200915-180f-448b-b135-1a24d5189953"
-MAGPROXY_AZURE_CLIENT_SECRET=""
+MAGPROXY_AZURE_CLIENT_SECRET="banana"
 MAGPROXY_AZURE_SCOPE="api://ec200915-180f-448b-b135-1a24d5189953/.default"
 
-RABBIT_MQ_URI="amqp://guest:guest@localhost:5672"
+# rabbitmq
+RABBIT_MQ_URI="amqp://guest:guest@localhost:5622"
+
+# always pass turnstile_secret - https://developers.cloudflare.com/turnstile/troubleshooting/testing/
+TURNSTILE_SECRET="1x0000000000000000000000000000000AA"
 
-TURNSTILE_SECRET=""
-MAILGUN_API_KEY=""
+# mailgun
+MAILGUN_API_KEY="banana"
+DEFAULT_MAILGUN_DOMAIN="banana"
 
-# nases
+# nases - get credentials to do eID user verification
 SLOVENSKO_SK_CONTAINER_URI="https://fix.slovensko-sk-api.bratislava.sk"
 API_TOKEN_PRIVATE="banana"
 OBO_TOKEN_PUBLIC="banana"
 SUB_NASES_TECHNICAL_ACCOUNT="banana"
 
 # bloomreach
-BLOOMREACH_INTEGRATION_STATE="ACTIVE"
-BLOOMREACH_PROJECT_TOKEN=""
-BLOOMREACH_API_KEY=""
-BLOOMREACH_API_SECRET=""
+BLOOMREACH_INTEGRATION_STATE="INACTIVE"
```

Is there a reason to turn Bloomreach OFF? I rather have it ON and get error messages if related env var are missing when in use. Not having this env vars should not stop application from running.

```suggestion
BLOOMREACH_INTEGRATION_STATE="ACTIVE"
```

---

As agreed in person - kept it off, but added logs to warn you that bloomreach interaction is inactive everytime an event is supposed to get tracked.

@Balros implied some further work intended here :). Leaving up to you to define what needs to be done.